### PR TITLE
add FloatBotton component open api description and demo

### PR DESCRIPTION
**Pull request by poyiding**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     add FloatBotton component open api description and demo      |
| 🇨🇳 Chinese |     补充  FloatBotton 组件 open api 说明和demo |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at be2b2b3</samp>

Updated the demo and documentation for the `FloatButton.Group` component to show how to use the `open` and `onOpenChange` props for controlling the menu visibility. Applied the changes to both English and Chinese versions of the documentation.

### <samp>🤖 Generated by Copilot at be2b2b3</samp>

*  Update `FloatButton.Group` demo code to include an example of controlling menu visibility with `open` and `onOpenChange` props ([link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-a750647d861179cb216e361729f6b01d2fb25ca3ee51ea6d5954a4832d62d0c9L1-R47))
*  Import `useState` hook and `QuestionCircleOutlined` icon for the new example ([link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-a750647d861179cb216e361729f6b01d2fb25ca3ee51ea6d5954a4832d62d0c9L1-R47))
*  Adjust style props to make room for the new example ([link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-a750647d861179cb216e361729f6b01d2fb25ca3ee51ea6d5954a4832d62d0c9L1-R47))
*  Clarify the usage of `open`, `onOpenChange`, and `trigger` props in `FloatButton.Group` documentation in both English and Chinese ([link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-80a107045d1a20a6bb129a0a9728f40fbdb0844205cc57fb4ea23271b99e8adcL57-R58), [link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-778117125c1b0d160ec5e0ff086b12c8118fdad8b97d28f351948704a1fe1e11L54-R59))
*  Leave version column blank for the props as they are not new features ([link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-80a107045d1a20a6bb129a0a9728f40fbdb0844205cc57fb4ea23271b99e8adcL57-R58), [link](https://github.com/ant-design/ant-design/pull/42835/files?diff=unified&w=0#diff-778117125c1b0d160ec5e0ff086b12c8118fdad8b97d28f351948704a1fe1e11L54-R59))

### CODE REVIEW

File: components/float-button/index.en-US.md [hunk 1]
```diff
@@ -54,8 +54,8 @@ FloatButton. Available since `5.0.0`.
 | --- | --- | --- | --- | --- |
 | shape | Setting button shape of children | `circle` \| `square` | `circle` |  |
 | trigger | Which action can trigger menu open/close | `click` \| `hover` | - |  |
-| open | Whether the menu is visible or not | boolean | - |  |
-| onOpenChange | Callback executed when active menu is changed | (open: boolean) => void | - |  |
+| open | Whether the menu is visible or not, use it with trigger | boolean | - |  |
```
Comment (kiner-tang) [inline components/float-button/index.en-US.md:57]: trigger 不是有默认值吗？也就是 trigger 始终都是有值的，那这里跟 trigger 配合使用是啥意思，怎么配合使用呢？比较难理解的样子

Comment (kiner-tang) [reply components/float-button/index.en-US.md:57]: image
哦，前面的 \| 看成 ｜ 了，以为默认值是 hover 🥲

Comment (poyiding) [reply components/float-button/index.en-US.md:57]: 当设置 trigger 属性时 open 才生效,  看这个 https://codesandbox.io/s/ji-ben-antd-5-5-2-forked-hwuioc?file=/demo.tsx
用户容易误解，怎么配合使用 我补充了demo的

File: components/float-button/demo/group-menu.tsx [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
-      style={{ right: 94 }}
-      icon={<CustomerServiceOutlined />}
-    >
-      <FloatButton />
-      <FloatButton icon={<CommentOutlined />} />
-    </FloatButton.Group>
-  </>
-);
+const App: React.FC = () => {
+  const [open, setOpen] = useState(true);
```
Comment (zombieJ) [inline components/float-button/demo/group-menu.tsx:10]: 这样会看起来像是个 bug，可以单独搞个 Demo，然后加个 Switch 切换展开~

File: components/float-button/demo/controlled.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  const onChange = (checked: boolean) => {
+    setOpen(checked);
+  };
+
+  return (
+    <>
+      <FloatButton.Group
+        open={open}
+        trigger="click"
```
Comment (zombieJ) [inline components/float-button/demo/controlled.tsx:16]: 这有点问题，实际上 click 是没用的，但是它却需要设置才能用。应该是 open 设置了就跟着 open 走才是受控

Comment (MadCcc) [reply components/float-button/demo/controlled.tsx:16]: 加上 `onOpenChange` 完整点

**Review by MadCcc — APPROVED** [review]

File: components/float-button/FloatButtonGroup.tsx [hunk 4]
```diff
@@ -88,6 +89,13 @@ const FloatButtonGroup: React.FC<FloatButtonGroupProps> = (props) => {
     }
   }, [trigger]);
 
+  // =================== Warning =====================
+  warning(
+    !('open' in props && !('trigger' in props)),
```
Comment (zombieJ) [inline components/float-button/FloatButtonGroup.tsx:94]:
```tsx
if (process.env.NODE_ENV !== 'production') {
  warning(typeof open !== 'boolean' || props.trigger, ...)
}
```

Comment (zombieJ) [reply components/float-button/FloatButtonGroup.tsx:94]: 不要用 in，HOC 可能会把 `undefined` 传进来

**Comment by zombieJ** [comment]
截屏2023-06-08 10 26 44

另一个 open onOpenChange 的例子有点 tricky，不看代码以为是坏了。这个 Switch 例子就够了，另一个 onOpenChange 受控例子可以不用写。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**